shared/test/test/PasscodeUITests.swift

```diff
@@ -0,0 +1,113 @@
+FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+
+import XCTest
+
+class PasscodeUITest: SalesforceNoSessionTestCase {
+    
+    let loginHelper = LoginHelper()
+    let loginPage = LoginPage()
+    let hostPage = HostPage()
+    let searchScreen = SearchScreen()
+    let userListScreen = UserListScreen()
+    let passcodePage = PasscodePage()
+    
+    // MARK: Setup
+    override func setUp() {
+        super.setUp()
+        XCTAssertNotNil(passcodeTimeout, "Expect passcode timeout defined")
+        XCTAssertNotNil(passcodeLength, "Expect passcode length defined")
+        print("Passcode:" + passcode)
+        loginPage.waitForPageLoaded()
+    }
+    
+    override func tearDown() {
+        loginPage.waitForPageLoaded()
+        super.tearDown()
+    }
+    
+    // MARK: Tests
+    func testPasscode() {
+        let user = accountWithPasscode.valueForKey("username") as! String
+        let password = accountWithPasscode.valueForKey("password") as! String
+        let host = accountWithPasscode.valueForKey("host") as! String
+        loginHelper.loginToSalesforce(user, password: password, url: host, withPasscode: passcode)
+        
+        //verify activity timeout
+        sleep(passcodeTimeout! - 10)
+        XCTAssertFalse(passcodePage.isPresented()) //should not present passcode
+        sleep(15)
+        XCTAssertTrue(passcodePage.isPresented() && passcodePage.getStatus()==PasscodeStatus.Verifying) //should present passcode
+        
+        //verify passcode enter
+        passcodePage.verifyPasscode(getDifferentString())
+        XCTAssertTrue(passcodePage.isPresented() && passcodePage.getStatus()==PasscodeStatus.Verifying) //should
+        passcodePage.verifyPasscode(getDifferentString())
+        XCTAssertTrue(passcodePage.isPresented() && passcodePage.getStatus()==PasscodeStatus.Verifying) //should
+        
+        passcodePage.enterPasscode(passcode)
+        backspace(1) //backspace
+        XCTAssertTrue(passcodePage.isPresented() && passcodePage.getStatus()==PasscodeStatus.Verifying)
+        passcodePage.enterPasscode(passcode.substringFromIndex(passcode.endIndex.advancedBy(-1)))
+        passcodePage.done()
+        XCTAssertFalse(passcodePage.isPresented()) //should not present passcode
+        
+        //TODO: verify background/foreground after timeout
+//        XCUIDevice().pressButton(XCUIDeviceButton.Home)
```

Comment:
ok, I should probably remove this part for now, I am hitting some issue when run app.resolve() which supposed to bring the app to foreground..